core/src/main/java/de/danoeh/antennapod/core/storage/FeedFileDownloadStatusRequesterInterface.java

```diff
@@ -0,0 +1,13 @@
+package de.danoeh.antennapod.core.storage;
+
+import android.support.annotation.NonNull;
+
+import de.danoeh.antennapod.core.feed.FeedFile;
+
+public interface FeedFileDownloadStatusRequesterInterface {
```

Comment:
Could this be done by mocking objects instead? I feel like this is changing too much of the actual logic just  for the tests.

Reply:
The interface is created for 2 reasons:
1. Makes it explicit that `ItemEnqueuePositionCalculator` really needs a very narrow subset of `DownloadRequester` (read-only) methods, e.g., it won't ever attempt to download.
2. Mock libraries could be used. I felt it first needs to be discussed, given it's never used in the project.
E.g., if we as a group are comfortable, I could add mockito, irrespective of whether `ItemEnqueuePositionCalculator` is kept or not.

Reply:
Mockito is already used in FeedMediaTest. I don't really care about additional (well-known) libraries for tests. They are not shipped to users, so they can't cause problems.

Reply:
What do you think about renaming to DownloadStateProvider or something like that? The current name feels like it is totally specific to this one usage.

Reply:
fixed - it's a better name indeed.